### ISSUE TITLE
Nelornieth embassy event chain

### ISSUE THREAD

**Issue by BriceLeduc** [jj248/RealmsInExile#748]
Nelornieth embassy decision event chain is functional and ready for testing, though the loc is still very much WIP. Here is how it should go : 

- Decide which kingdom to send an embassy to : Lasgalen, Lothlorien, 3rd random Elven kingdom if they exist (Valinor excluded, Elrond might pop up ?)
- Decide who will lead the embassy : any councillor or your heir, depending on their skills and traits
- Decide which activity to organise first : banquet, hunting, meditation, song performance
- Target ruler reacts to the activity (based on own and ambassador traits) -> result event for the initiator of the king of Nelornieth
- Decide which activity to organise next : banquet, hunting, meditation, song performance - previous activity excluded
- Target ruler reacts to the activity (based on own and ambassador traits) -> result event for the initiator of the king of Nelornieth
- Embassy over, according to the reaction of the target to the activities, different boons can be requested : alliance, weapon, a knight or nothing.
- If alliance requested, event for the target to decide (ai should say yes)

**Comment by savagesach** [comment]
I'm doing some preliminary testing and the decision name is still bugged
![image](https://user-images.githubusercontent.com/86676697/221998503-fee93dec-937f-4ee3-8048-929bdf47035e.png)

**Comment by savagesach** [comment]
also the decision does not do anything, it just takes your money and prestige and says there's no one available to ally.
![image](https://user-images.githubusercontent.com/86676697/221998835-d204076f-c27c-48ce-86be-2ae2d3a51d2a.png)

**Comment by jj248** [comment]
@BriceLeduc

**Comment by savagesach** [comment]
The decision does work now, so I'll test my progress from here
![image](https://user-images.githubusercontent.com/86676697/222202976-c2a8fc45-f0a6-4e68-9249-c694cbbbae2d.png)
![image](https://user-images.githubusercontent.com/86676697/222203028-6294eb7f-23c6-408c-bf80-ad86f9ab0e46.png)

**Comment by savagesach** [comment]
I've gone through the events and I like it so far. Aside from the obvious issues with not actually getting an alliance and localization,  it's really good. The only suggestion I'd make is to tone down the cost to maybe 750 gold and prestige, as I think 1000 is a bit much

**Comment by BriceLeduc** [comment]
The code part of the event is done, with 3 event locs left to write. Ready for testing.
I've reduced the cost to 700g and 1000 prestige, tell me what you think of its current state

**Comment by savagesach** [comment]
Now that I think about it, being able to use this decision should not depend on thranduil's situation alone. That's unfair considering the alliance offer applies to galadriel just as much as it does to thranduil. I think a better solution would be to not display galadriel or thranduil as options to ally in the later event if they are imprisoned.
![image](https://user-images.githubusercontent.com/86676697/222626278-40235873-5c67-47b2-9f05-c7943073901b.png)

**Comment by BriceLeduc** [comment]
Right, this comes from the fact I didn't make the decision myself, it's not fully in line with the event design

**Comment by BriceLeduc** [comment]
OK the chain is now fully implemented, please tell me what you think of it : 

- cost/reward balance
- success/neutral/failure chances (use debug mode to see AI weighting)
- general feel of the event chain

**Comment by savagesach** [comment]
![image](https://user-images.githubusercontent.com/86676697/224723315-33457f79-53f2-4fc0-b626-af68fc1085d6.png)
![image](https://user-images.githubusercontent.com/86676697/224723366-931cebe4-90f4-4cf7-8877-b7c4a62f599d.png)
I think choosing lorien should have the same cost as choosing thranduil

**Comment by savagesach** [comment]
I've tested the alliance further and it turns out even with it, thranduil and lanthir can't join each other's wars
![image](https://user-images.githubusercontent.com/86676697/224729338-1c8a751c-7bfc-4500-b40c-0b1e60743ab7.png)

**Comment by savagesach** [comment]
![image](https://user-images.githubusercontent.com/86676697/227022075-6bb12a40-f01f-4dba-951e-4e30f074faa5.png)
this works as intended

**Comment by Remerod** [comment]
*   Decide which kingdom to send an embassy to : Lasgalen, Lothlorien, 3rd random Elven kingdom if they exist (Valinor excluded, Elrond might pop up ?) 
 -Nemcraft seems to have tested this and it works
*    Decide who will lead the embassy : any councillor or your heir, depending on their skills and traits
 -Nemcraft seems to have tested this and it works
*    Decide which activity to organise first : banquet, hunting, meditation, song performance
 -Nemcraft seems to have tested this and it works
*    Target ruler gets an event to reacts to the activity (based on own and ambassador traits) -> result event for the initiator of the king of Nelornieth
 -Nemcraft seems to have tested this and it works
*    Decide which activity to organise next : banquet, hunting, meditation, song performance - previous activity excluded
 -Nemcraft seems to have tested this and it works
*    Target ruler gets an event to reacts to the activity (based on own and ambassador traits) -> result event for the initiator of the king of Nelornieth
 -Nemcraft seems to have tested this and it works
*    Embassy over, according to the reaction of the target to the activities, different boons can be requested : alliance, weapon, a knight or nothing.
 -Nemcraft seems to have tested this and it works
*    If alliance requested, event for the target to decide (ai should say yes)
 -Nemcraft seems to have tested this and it works
